src/utils.ml

```diff
@@ -88,8 +112,9 @@ module Ast_io = struct
         What we do instead is create a lexing buffer from the input channel and pre-fill
         it with what we read to do the test. *)
       let lexbuf = Lexing.from_channel ic in
-      let len = String.length prefix_read_from_source in
-      Bytes.blit_string ~src:prefix_read_from_source ~src_pos:0 ~dst:lexbuf.lex_buffer ~dst_pos:0
+      let pre_read = skip_hash_bang ic prefix_read_from_source in
+      let len = String.length pre_read in
+      Bytes.blit_string ~src:pre_read ~src_pos:0 ~dst:lexbuf.lex_buffer ~dst_pos:0
```

Comment:
What about first blitting `prefix_read_from_source` into the lexing buffer, and then calling a lexer that does the same as the `skip_hash_bang` in the compiler? The lexer is a bit easier to read the the OCaml parsing code.

Reply:
That's a good idea! I didn't know that was possible. I've named that new lexer `Lexer_internal` for now to not shadow the `Lexer` module we're exposing at the moment. I would rename `Lexer_internal` to `Lexer` in the PR in which I remove the current `Lexer` module.

Reply:
I've just noticed that renaming it to `Lexer` won't be possible, since we want to keep the possibility to access the compiler-libs `Lexer` through Ppxlib.

Reply:
TBH, i would call this module just `Skip_hash_bang`, since that's its only purpose

Reply:
Oh yes, that makes sense. I've renamed it.